# 1. 단일표본 T-test

T검정은 단일 또는 2개 집단의 평균 비교까지만 가능한 검증방법이다.
(대소를 비교하는 것: 단일, 등호 비교: 2개 집단)
"표본의 평균이 모집단의 평균과 일치한다"라는 귀무가설을 확인하는 방법이다.
<br>
**파이썬 내장 라이브러리 scipy의 ttest_1samp를 활용**

In [1]:
from scipy.stats import ttest_1samp

In [2]:
data = [ 32, 34 , 29 , 29 , 22 , 39, 38 , 37 , 38 , 36 , 30, 26 , 22 , 22]
# 모집단 평균이 이렇다고 가정
expected_mean = 30

In [4]:
tstat, pval = ttest_1samp(data, expected_mean)
print(tstat, pval)

0.5973799001456603 0.5605155888171379


In [5]:
# numpy를 이용해도 된다.
import numpy as np
data = np.array([ 32, 34 , 29 , 29 , 22 , 39, 38 , 37 , 38 , 36 , 30, 26 , 22 , 22])
expected_mean = 30

tstat, pval = ttest_1samp(data, expected_mean)
print(tstat, pval)

0.5973799001456603 0.5605155888171379


대부분의 연구에서는 p-value를 0.05로 잡는다. 만약 이보다 작다면 귀무가설을 기각하는 것이다.
본 표본의 p-value는 0.5605155888171379 이므로 제1종오류가 범해질 확률이 크다. 이는 곧 귀무가설이 타당하다고 여겨질 근거가 되며
마지노선인 0.05 아래로 내려가지 않으므로 귀무가설을 채택한다.

# 2. 독립표본 T-test

H0: "2개 집단에서 표본을 수집했을 때, 두 집단의 평균이 서로 일치한다"

In [6]:
from scipy.stats import ttest_ind

data1 = [ 32, 34 , 29 , 29 , 22 , 39, 38 , 37 , 38 , 36 , 30, 26 , 22 , 22]
#data2 = [ 32, 34 , 29 , 29 , 22 , 39, 38 , 37 , 38 , 36 , 30, 26 , 22 , 22] 귀무가설 채택용 데이터
data2 = [ 132, 134 , 129 , 129 , 122 , 139, 138 , 137 , 138 , 136 , 130, 126 , 122 , 122]

tstat, pval = ttest_ind(data1, data2)

print(tstat, pval)
ttest_ind(data1, data2)

-42.24113783375391 1.7260271771784527e-25


Ttest_indResult(statistic=-42.24113783375391, pvalue=1.7260271771784527e-25)

통계검정량 t 값은 -42.24113783375391, 이에 대한 pvalue는 1.7260271771784527e-25 이다.
즉 제1종오류가 범해질 확률이 1.7260271771784527e-25로 마지노선인 0.05를 넘겼다. 제1종오류가 발생할 확률이 현저히 낮으며 이는 역으로 귀무가설 기각에 힘을 실어주는 통계적 근거이다. <br>
즉, 표본 data1, data2 두 집단의 평균이 서로 다르다.

In [8]:
import researchpy as rp
import numpy as np
import pandas as pd

In [11]:
df_y = pd.DataFrame(np.array(data1), columns = ['y'])
df_yhat = pd.DataFrame(np.array(data2), columns = ['yhat'])

y = df_y['y']
yhat = df_yhat['yhat']
print(y.shape, yhat.shape)
print(y, yhat)

(14,) (14,)
0     32
1     34
2     29
3     29
4     22
5     39
6     38
7     37
8     38
9     36
10    30
11    26
12    22
13    22
Name: y, dtype: int32 0     132
1     134
2     129
3     129
4     122
5     139
6     138
7     137
8     138
9     136
10    130
11    126
12    122
13    122
Name: yhat, dtype: int32


In [12]:
summary, rst = rp.ttest(group1=y, group1_name='Y'
                       ,group2=yhat, group2_name='linear interpolation')

summary

Variable     N   Mean         SD        SE   95% Conf.  \
0                     Y  14.0   31.0   6.263447  1.673977   27.383593   
1  linear interpolation  14.0  131.0   6.263447  1.673977  127.383593   
2              combined  28.0   81.0  51.287137  9.692358   61.112925   

     Interval  
0   34.616407  
1  134.616407  
2  100.887075

In [13]:
rst

Independent t-test   results
0  Difference (Y - linear interpolation) =  -100.0000
1                     Degrees of freedom =    26.0000
2                                      t =   -42.2411
3                  Two side test p value =     0.0000
4                 Difference < 0 p value =     0.0000
5                 Difference > 0 p value =     1.0000
6                              Cohen's d =   -15.9656
7                              Hedge's g =   -15.5006
8                          Glass's delta =   -15.9656
9                            Pearson's r =     0.9928